_source/_posts/2022-01-18-deploy-dotnet-container-aws-fargate.md

```diff
@@ -0,0 +1,640 @@
+
+```bash
+docker build --build-arg CERT_PASSWORD=mypass123 -t okta-chat .
+```
+
+Select the copy button in the URI column next to your image and save that for later. This is the path to your image.
+
+{% img blog/dotnet-aws-ecs-fargate/18-oktaChatRepo.png alt:"AWS save URI" width:"800" %}{: .center-image }
+
+### Set up the ECS Fargate Cluster
+
+Now set up your ECS cluster.
+
+On the left menu, click **Clusters** under **Amazon ECS**, then click the **Create Cluster** button.
+
+Click **Networking Only** > **Next step**from the options.
+
+{% img blog/dotnet-aws-ecs-fargate/19-NetworkOnly.PNG alt:"AWS Networking Only" width:"800" %}{: .center-image }
+
+I named my cluster "okta-test", name yours as you see fit, and click **Create** on the following screen click **View Cluster**.
+
+Now that the cluster is set-up you'll need to create a task for your image to run as.
+
+Click **Task Definitions** on the left menu and click **Create new Task Definition**.
+
+For launch type, select **FARGATE** and click **Next step**.
+
+{% img blog/dotnet-aws-ecs-fargate/20-LaunchType.PNG alt:"Fargate launch type" width:"800" %}{: .center-image }
+
+For the **Task Definition Name**, I named mine **okta-chat**.
+
+Set the **Task memory (GB)** to 0.5 GB.
+Set the **Task CPU** to 0.25 vCPU.
+
+.NET Core applications are very efficient, as are containers. For many applications, you'll find you can serve many requests with smaller boxes than you might typically be accustomed to.
+
+{% img blog/dotnet-aws-ecs-fargate/21-TaskSize.PNG alt:"Task size" width:"800" %}{: .center-image }
+
+Leave other options unchanged.
+
+Now you need to define the image you want to use. Click **Add Container**.
+
+I'll name the container the same thing **okta-chat**.
+
+For the **Image**, you'll need the path you copied from the ECR repository earlier.
+
+Set the soft limit to 512, and add ports 80 and 443.
+
+{% img blog/dotnet-aws-ecs-fargate/22-AddContainer.PNG alt:"Add container ports" width:"800" %}{: .center-image }
+
+Scroll down to the **ENVIRONMENT** section and add the following key-value pairs:
+
+```bash
+ASPNETCORE_URLS="https://+;http://+"
+ASPNETCORE_HTTPS_PORT=443
+ASPNETCORE_Kestrel__Certificates__Default__Password="mypass123"
+ASPNETCORE_Kestrel__Certificates__Default__Path=./aspnetapp.pfx
+```
+
+{% img blog/dotnet-aws-ecs-fargate/27-add-env-vars.png alt:"Run new task" width:"800" %}{: .center-image }
+
+Click **Add** at the bottom of the popup. Now click **Create** at the bottom of the page. Go back to your cluster, click on the **Tasks** tab, and click **Run new Task**.
+
+> **Note**: It is recommended to create services instead of tasks for production so that you take full advantage of the elastic scaling capabilities of ECS. I'm using tasks directly for the simplicity of the demo. Using services is also required when using an ALB to route traffic to the application.
```

Comment:
```suggestion
> **Note**: Here's a recommendation: Create services instead of tasks for production to take full advantage of the elastic scaling capabilities of ECS. I'm using tasks directly for the simplicity of the demo. When using an ALB (application load balancer) to route traffic to the application, use of services is also required.
```
avoids passive voice. spells out the acronym